src/map/lua/lua_baseentity.cpp

```diff
@@ -4402,7 +4402,17 @@ inline int32 CLuaBaseEntity::canUsePet(lua_State *L)
 {
     DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
 
-    lua_pushboolean(L, m_PBaseEntity->loc.zone->CanUseMisc(MISC_PET));
+    if (dynamic_cast<CCharEntity*>(m_PBaseEntity) == nullptr)
```

Comment:
almost! can tidy up a bit by doing this instead
```
auto PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity);
if (PChar && PChar->loc.zone->CanUseMisc(MISC_PET) && !PChar->m_moghouseID)
    lua_pushboolean(L, true);
else
    lua_pushboolean(L, false);

return 1;
```
if it fails on casting to CCharEntity* we know its not a player, and this check is always false for mobs so return false
if the cast is successful, we know its a player so we check if the zone allows pets and they player is not in mog house